### PR TITLE
Exercise 3

### PR DESCRIPTION
**Pull request by samyou-softwire**

### CODE REVIEW

File: todo_app/data/view_model.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+
+class ViewModel:
+    def __init__(self, tasks: list[Task]):
+        self.tasks = tasks
+
+    def incomplete_items(self):
+        return [task for task in self.tasks if task.status == 'incomplete']
+
+    def complete_items(self):
```
Comment (AgathiyanB) [inline todo_app/data/view_model.py:11]: Optionally could include decorator @property

File: todo_app/app.py [hunk 2]
```diff
@@ -1,46 +1,49 @@
...
-        task.status = toggle(request.form.get("status"))
+        if submit_type in ["changestatus", "update"]:
+            task.description = request.form.get("description")
 
-    if submit_type in ["changestatus", "update"]:
-        task.description = request.form.get("description")
+            due = request.form.get("due")
+            if due == "":
+                due = None
+            task.due = dateparser.parse(due) if due is not None else None
```
Comment (AgathiyanB) [inline todo_app/app.py:41]: Instead of `if due is not None else` you can write `if due else`

File: tests/data/view_model_test.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+def test_items_status(simple_view_model, get_tasks: Callable[[ViewModel], list[Task]], expected: str, rest: str):
+    tasks = get_tasks(simple_view_model)
+
+    assert all(task.status == expected for task in tasks)
+
+    # subtract complete tasks from tasks to get the rest of the tasks
+    other_tasks = [task for task in simple_view_model.tasks if task not in tasks]
+
+    # so every other task should have status incomplete
+    assert all(task.status == rest for task in other_tasks)
```
Comment (AgathiyanB) [inline tests/data/view_model_test.py:34]: All your tests look really good

**Review by AgathiyanB — COMMENTED** [review]
Pretty good, nothing really much I'd change

**Review by CharlieCumber1 — APPROVED** [review]
Great work getting the unit and integration tests working - I think they test the correct things.

Feel free to merge this once you have considered my comments.

**Pull request merged**